### DISCUSSION TITLE
How to package your code to share with others

### DISCUSSION THREAD

**Discussion by gavlig** [custom-components/pyscript#182] [Q&A] [answered]
Hey, i want to share a small framework i wrote with my friend and i'm a bit confused about that.
What i have right now:

```
pyscript/
    bedroom.py
    living_room.py
    modules/
        area_controller.py
```

With that setup when i do `from area_controller import area_controller` from my little scripts (bedroom.py and living_room.py) they work and make me happy. However I wanted to share area_controller.py as a git repo but i can't suggest to clone my repo as "modules" folder so i thought i'd put it into a separate folder, but after doing that i can't import area_controller :/ (ModuleNotFoundError) Tried `from my_repo_name import area_controller` and just `import area_controller` I'm not super experienced with python, so i might be missing something obvious here but i can't find what exactly.

Also tried to use apps from these docs with the same outcome:
https://hacs-pyscript.readthedocs.io/en/latest/reference.html

Any examples of working apps or modules will be greatly appreciated! :)

**Comment by craigbarratt** [comment, marked as answer]
You should turn your module into a package by putting it in a subdirectory:
```
pyscript/
    bedroom.py
    living_room.py
    modules/
        area_controller/
            __init__.py
```
The key thing is that in a package, the main file is always called `__init__.py`.  Then if you want to share your `area_controller` package you would share the directory `pyscript/modules/area_controller`.

**Reply by gavlig** [reply]
Thank you, that helped! I was too stubborn to rename my file for some reason XD
